devtools/x/src/lint/guppy.rs

```diff
@@ -146,3 +146,51 @@ impl PackageLinter for CrateNamesPaths {
         Ok(RunStatus::Executed)
     }
 }
+
+/// Ensure libra-workspace-hack is a dependency
+#[derive(Debug)]
+pub struct WorkspaceHack;
+
+impl Linter for WorkspaceHack {
+    fn name(&self) -> &'static str {
+        "workspace-hack"
+    }
+}
+
+impl PackageLinter for WorkspaceHack {
+    fn run<'l>(
+        &self,
+        ctx: &PackageContext<'l>,
+        out: &mut LintFormatter<'l, '_>,
+    ) -> Result<RunStatus<'l>> {
+        let package_id = ctx.metadata().id();
+        let pkg_graph = ctx.project_ctx().package_graph().expect("can't find package graph");
+        let workspace_hack_id = pkg_graph.packages().find_map(|metadata| {
+            if metadata.name() == "libra-workspace-hack" {
+                return Some(metadata.id());
+            }
+            None
```

Comment:
You can search through just the workspace packages for a cheaper query. (I have a task filed to add a by-package-name index.)

Reply:
actually, you can just search through the links of each package if you like as well.

Reply:
`pkg_graph.dep_links(package.id()).unwrap()` should give you a list of links, and then you can look at `to.name()` to see if it's correct.

Reply:
Done.